.golangci.yaml

```diff
@@ -13,6 +13,16 @@ linters-settings:
       # We want to enforce all values are specified when inserting or updating
       # a database row. Ref: #9936
       - 'github.com/coder/coder/v2/coderd/database\.[^G][^e][^t]\w+Params'
+
+  forbidigo:
+    forbid:
+      # Usage of the http package global DefaultClient and the associated
+      # functions that wrap said global are known to cause flakiness with
+      # parallel tests. They also introduce global state to production code.
+      # Prefer isolated instances of an http.Client.
+      - {p: '^http\.(Get|Post|Head|PostForm)(\(|$)', msg: 'methods wrap http.DefaultClient which introduces global state. Use a http.Client with timeouts.'}
```

Comment:
AFAIK `http.Client` still uses a global `http.DefaultTransport` by default as well.
We probably want to force a fresh transport too, or prefer the use of https://github.com/hashicorp/go-cleanhttp.

Reply:
Also, the new clients you're creating do not actually specify custom timeouts, which you're admonishing the reader to do when receiving this linter error. Maybe we need a helper which sets up some reasonable defaults with no shared state?

Reply:
Yep you're right `CloseIdleConnections` effectively applies to the `http.DefaultTransport`. I'm looking into isolating the client transports. That might be a good opportunity to create a helper.

Fair point on the timeout messaging. Honestly, I don't have confidence in choosing a timeout for the existing code. I mainly wanted that error messaging to at least make the reader _think_ about it in the context of the code being written.